src/EditorFeatures/TestUtilities/LanguageServer/AbstractLanguageServerProtocolTests.cs

```diff
@@ -238,12 +239,13 @@ protected static LSP.CompletionParams CreateCompletionParams(LSP.Location caret,
                 SortText = text,
                 InsertTextFormat = LSP.InsertTextFormat.Plaintext,
                 Kind = kind,
-                Data = new CompletionResolveData()
+                Data = JObject.FromObject(new CompletionResolveData()
                 {
                     DisplayText = text,
                     TextDocument = requestParameters.TextDocument,
                     Position = requestParameters.Position
-                },
+                }),
+                Icon = tags != null ? new ImageElement(tags.ToImmutableArray().GetFirstGlyph().GetImageId()) : null,
```

Comment:
Not using tags.ToImmutableArray().GetFirstGlyph().GetImageElement()? You added the extension method... :smile:

Reply:
🤦‍♂️
I think I need to start this PR again.. its been rebased so many times to different branches I don't even know what is real any more